app/assets/javascripts/auto_title.js

```diff
@@ -7,7 +7,7 @@ function path(name) {
 }
 
 function retrieveTitleSuggestion() {
-  var link = this.value;
+  var link = encodeURI(this.value);
```

Comment:
Encoding the URL on the client side doesn't change the security situation b/c the user can do whatever they want with your javascript. someone could very easily open the development console of their browser and remove the call to `encodeURI` if they so choose. Protection against XSS needs to happen on the server, which, like you mentioned, is being sure to sanitize all strings that you send back to the client for display.